### DISCUSSION TITLE
Autoupdate

### DISCUSSION THREAD

**Discussion by mrJudahBella** [mattmc3/antidote#203] [Q&A] [answered]
Hi there! I'm new to Antidote, understand that there is a `antidote update` command for updating antidote and plugins _manually_. Is there a way to run this automatically once every few days or at launch?

**Comment by mattmc3** [comment, marked as answer]
There are a few different ways to do what you're looking for. If you have a desktop machine that's always on and you want updates in the background, you could add a cron entry to run your updates at a specific time every week.

For example, you can run `crontab -e` and add an entry for Monday at noon:
```
0 12 * * 1 /path/to/a/script/that/runs/antidote_update
```

If you are on a laptop and are okay with updates running once a week when you launch your shell, you could also simply add something like this to your .zshrc:

```zsh
# .zshrc stuff before this...

# Init antidote
source /path/to/antidote

# Look for a file telling us it's been over a week since we last updated.
# If found, delete it so we know to run antidote update.
find $(antidote home) -name '.up2date' -maxdepth 1 -type f -mtime +7 -delete
if [[ ! -r $(antidote home)/.up2date ]]; then
  antidote update
  touch $(antidote home)/.up2date
fi

# After updating if necessary, load your plugins:
antidote load

# more .zshrc stuff after this...
```

**Reply by mrJudahBella** [reply]
Thanks! I tried the cron route at first but didn't manage to get it to work. Don't know how to write a script that will run my zsh shell as me to run the `antidote update` command.. Would you be able to help advise?

I tried the following:

```
#!/bin/zsh

antidote update
```

Testing it with `sudo su <my username> -c ~/antidote-update` threw the following error: `antidote-update:3: command not found: antidote`

I messed around in `~/.config/zsh/.antidote/functions` and `chmod +x antidote-update` and ran that instead and it threw a whole bunch of other errors.. so I gave up.

FWIW I'm running the kickstart version of antidote. 

EDIT: I might just follow your .zshrc example instead as it's cleaner. Thanks!

**Reply by mattmc3** [reply]
If you need help with a script for a cron job, it's as simple as this:

```zsh
#!/usr/bin/env zsh
source /path/to/antidote.zsh
antidote update
```

Don't forget to `chmod` your script to give execute permissions (eg: `chmod +x /path/to/update_antidote_script`).

**Reply by mrJudahBella** [reply]
Wow thanks! I'll try that! 🙏
